File: .github/hooks/scripts/orchestrated_stop.py
from orchestrated_common import append_session_log, load_hook_input, load_state, print_json, repo_root_from_input


def main() -> int:
    input_data = load_hook_input()
    repo_root = repo_root_from_input(input_data)
    state = load_state(repo_root)

    if input_data.get("stop_hook_active", False):
        return print_json({})

    append_session_log(
        repo_root,
        {
            "sessionId": input_data.get("sessionId"),
            "conversationState": state.get("conversationState"),
            "activeTaskName": state.get("activeTaskName"),
            "queuedTasks": len(state.get("taskQueue", [])),
            "feedbackCount": len(state.get("feedbackLog", [])),
            "toolAuditCount": len(state.get("toolAudit", [])),
            "stopSignal": state.get("stopSignal", False),
            "lastUpdated": state.get("lastUpdated"),
        },
    )

    return print_json({})


if __name__ == "__main__":
    raise SystemExit(main())

File: .github/hooks/scripts/orchestrated_stop.mjs
import {
  appendSessionLog,
  archiveCurrentSpecBundle,
  loadHookInput,
  loadSessionState,
  loadState,
  printJson,
  repoRootFromInput,
  runCli,
  saveSessionState,
  saveState,
} from './orchestrated_common.mjs';

export async function main() {
  const inputData = await loadHookInput();
  const repoRoot = repoRootFromInput(inputData);
  const state = await loadState(repoRoot);
  const sessionState = await loadSessionState(repoRoot);
  const stopHookActive = inputData.stop_hook_active === true;

  if (stopHookActive) {
    return printJson({});
  }

  const archivedBundle = await archiveCurrentSpecBundle(repoRoot, state, sessionState);
  const archivedSpecPath = archivedBundle?.archivedFolderPath ?? null;

  state.currentSpecFolderPath = null;
  state.currentSpecPath = null;
  state.currentTaskListPath = null;
  sessionState.currentSpecFolderPath = null;
  sessionState.currentSpecPath = null;
  sessionState.currentTaskListPath = null;
  sessionState.lastArchivedSpecPath = archivedSpecPath;
  sessionState.status = 'stopped';
  sessionState.tasks = [];

  await saveState(repoRoot, state);
  await saveSessionState(repoRoot, sessionState);

  await appendSessionLog(repoRoot, {
    sessionId: inputData.sessionId,
    conversationState: state.conversationState,
    activeTaskName: state.activeTaskName,
    archivedSpecPath,
    toolAuditCount: state.toolAudit.length,
    stopSignal: state.stopSignal,
    lastUpdated: state.lastUpdated,
  });

  return printJson({});
}

await runCli(main);